pkg/network/driver/nmstate/integration_test.go

```diff
@@ -0,0 +1,449 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2022 Red Hat, Inc.
+ *
+ */
+
+package nmstate_test
+
+import (
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+
+	vishnetlink "github.com/vishvananda/netlink"
+
+	"kubevirt.io/kubevirt/pkg/network/driver/nmstate"
+	"kubevirt.io/kubevirt/pkg/network/driver/procsys"
+	"kubevirt.io/kubevirt/pkg/pointer"
+)
+
+var runIntegrationTests bool
+
+func init() {
+	flag.BoolVar(&runIntegrationTests, "run-integration-tests", false, "run integration tests")
+}
+
+var _ = ginkgo.Describe("NMState", integrationLabel, func() {
+	ginkgo.BeforeEach(func() {
+		if !runIntegrationTests {
+			ginkgo.Skip("integration tests are not set to run")
+		}
+	})
+
+	var nmState = nmstate.New()
+
+	ginkgo.It("reports actual network devices", func() {
+		status, err := nmState.Read()
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+		gomega.Expect(status.Interfaces).NotTo(gomega.HaveLen(0))
+		gomega.Expect(status.Routes.Running).NotTo(gomega.HaveLen(0))
```

Comment:
Same

Reply:
Done

Reply:
Thank you for the change.
It was made in this [commit](https://github.com/kubevirt/kubevirt/pull/9064/commits/015befb54e364bf95029a4fa59778d76db904b09), but it could be moved to an earlier [commit](https://github.com/kubevirt/kubevirt/pull/9064/commits/0d2c1368242ceb4f83c20536c6c2958c00973e0d).